lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -222,12 +202,17 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
             }
 
         """
+        print("es_json === ", es_json)
         dir_list = []
         file_list = []
         for toc in es_json["hits"]["hits"]:
-            if toc["_source"]["parent"] == self.parent:
-                dir_list.append(toc["_source"].get("name", "None"))
-            if toc["_source"]["directory"] == self.parent:
-                file_list = toc["_source"].get("files", "None")
+            if (
+                toc["_source"]["parent"] == context["parent"]
+                and "name" in toc["_source"]
+            ):
+                dir_list.append(toc["_source"]["name"])
+            if toc["_source"]["directory"] == context["parent"]:
+                # Retrieve files list if present else add an empty list.
+                file_list = toc["_source"].get("files", [])
```

Comment:
I think this would be better if you reversed the order of these two `if` statements, and made the second be an `elif`.